src/Arcus.EventGrid.Publishing/EventGridPublisher.cs

```diff
@@ -23,13 +25,19 @@ namespace Arcus.EventGrid.Publishing
     /// <seealso cref="IEventGridPublisher"/>
     public class EventGridPublisher : IEventGridPublisher
     {
-        private readonly AsyncPolicy _resilientPolicy;
+        private const string DefaultAuthenticationHeaderName = "aeg-sas-key";
+
+        private readonly AsyncPolicy _resilientPolicy = Policy.NoOpAsync();
         private readonly string _authenticationKey;
-        private readonly ILogger _logger;
+        private readonly HttpClient _injectedHttpClient;
+        private readonly string _authenticationKeySecretName;
+        private readonly ISecretProvider _secretProvider;
+        private readonly IHttpClientFactory _httpClientFactory;
         private readonly EventGridPublisherOptions _options;
+        private readonly ILogger _logger;
 
+        private static readonly HttpClient DefaultHttpClient = new HttpClient();
```

Comment:
I might have asked this in the past, but shouldn't we inject HttpClient instead ?
(I do see an `_injectedHttpClient` member though)

Reply:
Yes, this PR is actually to answer that question. With these changes, we allow both (you know, to be compatible and not breaking).